packages/terralib/tests/shapefile/alternative/Layer.lua

```diff
@@ -64,7 +64,15 @@ return {
 			}
 		end
 		unitTest:assertError(indexDefaultError2, defaultValueMsg("index", true))
-		-- // SPATIAL INDEX
+
+		local epsgError = function()
+			Layer{
+				project = proj,
+				name = "Elevation",
+				file = filePath("cabecadeboi-box.shp", "terralib")
+			}
+		end
+		unitTest:assertError(epsgError, "It was not possible to find the projection of layer 'Elevation'. It should be one of the projections available at www.terrame.org/projections.html")
```

Comment:
The memory leak happens because you change the test logic, so, the behavior as well. The customWarning() does not stop the flow and createLayer() runs to the end deallocating pointers, because it is not necessary a layer to have a srid. This warning must be a warning, not a error.